### ISSUE TITLE
Increase in number of connections

### ISSUE THREAD

**Issue by bhisey** [spring-projects/spring-data-cassandra#1251]
Upgrading from spring boot 2.2.8 to 2.5.5. , we saw an increase in the number of connections ie, double what is configured in pool.local.size. 
We could not reproduce this issue only using the Cassandra driver upgrade and reproduced it with the Sping boot upgrade. We believe the increase in the number of connections is coming from the following code related to spring boot, which was added as a part upgrade from spring boot 2.2.8 to 2.5.5. We believe this code piece is added to initialize keyspace and believe it spawns an additional connection, which is mostly idle after the initial connection and is only kept open to destroy script on shutdown etc...

https://github.com/spring-projects/spring-data-cassandra/blob/3.2.5/spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CqlSessionFactoryBean.java#L447-L462

Can you please let us know why after the upgrade double the number of connections is needed and what if there is any possible impact of it?

**Comment by bhisey** [comment]
Let me know if you need any additional detail but its pretty easy to reproduce.

**Comment by mp911de** [comment]
Spring Boot 2.2.x has used the Datastax 3.x driver whereas Boot 2.3++ uses the Datastax 4.x driver. With the rewrite of driver internals, we had to rewrite our schema support and due to driver restrictions, we had to switch to a different allocation scheme for resources. Spring Boot uses specifically [`SessionFactoryFactoryBean`](https://github.com/spring-projects/spring-data-cassandra/blob/f2ae546c23930507b2d6d85f024e0f444c561416/spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/SessionFactoryFactoryBean.java#L124-L137) to perform the keyspace initialization.
